packages/firestore/test/integration/util/events_accumulator.ts

```diff
@@ -28,8 +28,14 @@ export class EventsAccumulator<
   private events: T[] = [];
   private waitingFor: number;
   private deferred: Deferred<T[]> | null = null;
+  private rejectAdditionalEvents = false;
 
   storeEvent: (evt: T) => void = (evt: T) => {
+    if (this.rejectAdditionalEvents) {
+      throw new Error(
+        'Additional event detected after assertNoAdditionalEvents called'
```

Comment:
To make this more actionable, I'd include the event... Something like:
```
    throw new Error(
      'Additional event detected after assertNoAdditionalEvents called:'
      JSON.stringify(evt)
    );
```

Reply:
Done.